meetings/2021-04-28.md

```diff
@@ -0,0 +1,358 @@
+# 2021-04-28 Solid Authorization
+
+* [Meeting](https://meet.jit.si/solid-authorization)
+* [Chat](https://gitter.im/solid/authorization-panel?at=6088092ac1a9210b3c23a55a)
+* [Notes](https://hackmd.io/mFgvpYMZR7Co_DIDgUaNwA)
+
+## Agenda
+
+* Minutes https://github.com/solid/authorization-panel/pull/207
+* Cleanup minutes: standard filename and extension https://github.com/solid/authorization-panel/pull/208
+* Add issue template https://github.com/solid/authorization-panel/pull/205
+* Commit [PR 201](https://github.com/solid/authorization-panel/pull/201)?
+* AnyOf, AllOf, etc...
+
+
+## Present
+
+* Justin Bingham
+* Eric Prud'hommeaux
+* Henry Story
+* Matthieu Bosquet
+* Sarven Capadisli
+
+
+## Minutes
+
+1. Commited Meeting Cleanups
+2. Appropve minutes from 207 without the ACP code discused in the last few minutes in the last call
+3. Elf Accept changed and then Commit on PR 205
+4. Merged 201
+
+
+## ACP apply policy algorithm
+
+**Henry**: Last week, we discussed a PR I made to 
+[the wac-acp-diff story](https://github.com/solid/authorization-panel/blob/main/proposals/wac-acp-diff-story.md) which we just committed.
+We looked at how the ACL ontology allows a behavior similar to ACP by creating an inferred  `:authorizes` relation 
+defined as:
+
+```Turtle
+:authorizes owl:inverseOf [
+    owl:propertyChainAxiom(:accessControl acl:accessTo)
+]
+```
```

Comment:
For those following here, I recommend looking at the [OWL 2 Primer §Property Chain](https://www.w3.org/TR/2012/REC-owl2-primer-20121211/#Property_Chains) section. You can get the Turtle version of each code snippet if you go to the top of the page and click a "Turtle Syntax" button. That will give you the definition for :hasGrandparent in Turtle as:

```Turtle
:hasGrandparent  owl:propertyChainAxiom  ( :hasParent  :hasParent ) .
```

(It is annoying that one cannot pass that decision around as a URL and I [sent a mail to the semantic-web mailing list](https://lists.w3.org/Archives/Public/semantic-web/2021May/) reporting that.) 

We also have the definition given in [OWL2 RDF Based Semantics](https://www.w3.org/TR/2012/REC-owl2-rdf-based-semantics-20121211/#table-semcond-chains).

I was working with this picture in mind:

<img src="https://user-images.githubusercontent.com/124506/116776491-f5144880-aa68-11eb-9a97-5cb0ae7b5420.jpg" width="35%">

But I wrote it out too quickly and got the order of the relations wrong. 

```Turtle
:authorizes owl:inverseOf [
    owl:propertyChainAxiom  (acl:accessTo :accessControl )
]
```

But it is still a good question whether even that is correct.
And indeed the following does not look right:

<img src="https://user-images.githubusercontent.com/124506/116778310-6bb54400-aa71-11eb-84b6-b335168d8da0.jpg" alt="counterexample" width="60%">

I can follow the `:accessTo` link from `<fake.acl>` follow the `:accessControl` relation from there and that does not give me an `:authorizes` link.

So yes, I should go back to the original proposal I made:

```Turtle
acl:accessTo owl:inverseOf [
        owl:propertyChainAxiom ( acl:accessControl :authorizes )
      ] .
```

Reply:
The above is a very good example of why we want to build on formally specified languages such as OWL, as it allows one to  make precise arguments as @matthieubosquet just made above.

Reply:
> I can follow the :accessTo link from <fake.acl> follow the :accessControl relation from there and that does not give me the :authorizes link that gives Control rights I started off with.

I'm not sure I see why the `:authorizes` statement cannot be inferred using that graph and the previous axiom.
```turtle
# If
ex:authorization1 acl:accessTo ex:resource .
ex:resource acl:accessControl <card.acl> .
# Then
<card.acl> :authorizes ex:authorization1 .
# Could we not also say
<fake.acl#authorization2> acl:accessTo ex:resource .
# Therefore
<card.acl> :authorizes <fake.acl#authorization2> .
# ?
```

Reply:
If I understand you correctly you have this picture in mind, where I have labelled the nodes. 
The mapping from your code is 
`ex:authorization1` -> `a1`
`ex:resource` -> `r`
`<fake.acl#authorization2>` -> `a2`

<img src="https://user-images.githubusercontent.com/124506/116780538-b937ae00-aa7d-11eb-9b5e-1e1031fac553.jpg" alt="my picture of mattieux's triples above" width="65%">

So I was trying to say that with the improved but still rejected rule
```turtle
:authorizes owl:inverseOf [
    owl:propertyChainAxiom  (acl:accessTo :accessControl )
]
```
we then get the dotted line representing `<card.acl#> :authorizes <fake.acl#a2>` which is not what is desired.
That would make it easy to create fake authorizations. 

But that no longer follows from
```Turtle
acl:accessTo owl:inverseOf [
        owl:propertyChainAxiom ( acl:accessControl :authorizes )
      ] .
```

Reply:
Thanks Henry, that makes sense. So, to summarize:
- The axiom allowing inference of the `acl:accessTo` statement seems safe
- The axiom allowing inference of the `acl:authorizes` statement is not safe because it introduces a problem in controlling the boundaries of the authorization graph applying to a resource X.

Given a set of server managed statements (by that I mean statements that "lawfully" belong in the authorization graph or in other words, assuming that the statements in the form `:x acl:accessTo :y .` are all part of the authorization graph and apply to the resource already), the `acl:authorizes` inferrence axiom could be used.

Therefore, it seems unlikely the axiom for inferring `acl:authorizes` could be used in production to maintain a compatibility layer, it may in certain circumstances be used to migrate from one authorization system to the other.

Reply:
Ah, thanks. What I thought was a mistake might have a use after all!

So yes, to first address your second bullet point, the rule for inferring `:authorizes`

```
:authorizes owl:inverseOf [
    owl:propertyChainAxiom  (acl:accessTo :accessControl )
]
```

could be used by new clients (ACP, WAC+, whatever...) on existing deployments. Or at least it allows us to see how existing deployments fit in with the proposal for an `:authorizes` relation.
Existing deployments do not show the `:authorizes` relation, but it is as if there always had been one such relation from the document to each `Authorization`. I.e. they could infer the blue dotted lines below.

<img alt="Current ACL with :authorizes inferencing" src="https://user-images.githubusercontent.com/124506/116784307-e55e2980-aa93-11eb-9b6c-19adaf364db1.jpg" width="65%">

But as shown in my previous comment, it is essential for the `Authorization` nodes to appear inside the document linked to by the `acl:accessControl` header. So there has to be some context around the `:authorizes` rule above that makes that clear. That could be done in N3. 

To your first bullet point, the rule for `accessTo` is on the other hand completely general.

```Turtle
acl:accessTo owl:inverseOf [
        owl:propertyChainAxiom ( acl:accessControl :authorizes )
      ] .
```      

 It can be used to help clients written for the acl ontology to continue making sense of servers producing ACP, WAC+ or whatever we come up with, by injecting the new triple in the Graph before the UI ACL code gets to run...

In any case both of these rules help show how the two systems relate, and increase our understanding of what is going on.

Reply:
Henry, out of curiosity, what do you use for those graphs?
It's quite handy and you seem very quick to make them.

Reply:
I have been using [Omnigraffle](https://www.omnigroup.com/omnigraffle/new/) for close to 16 years now. 
(I am not sure if there are new better tools out there)